benchalerts/workflows.py

```diff
@@ -115,13 +115,13 @@ def update_status(description, state, details_url):
     # If anything in here fails, we can!
     try:
         conbench = conbench or ConbenchClient()
-        all_comparisons, _ = conbench.get_comparison_to_baseline(
+        comparisons = conbench.get_comparison_to_baseline(
             contender_sha=contender_sha, z_score_threshold=z_score_threshold
         )
-        regressions = benchmarks_with_z_regressions(all_comparisons)
+        regressions = benchmarks_with_z_regressions(comparisons)
         log.info(f"Found the following regressions: {regressions}")
 
-        if not all_comparisons:
+        if not any(comparison.baseline_info for comparison in comparisons):
```

Comment:
Checking my understanding because this line keeps getting repeated: The move from `all_comparisons` to `not any(...)` is because `.get_comparison_to_baseline()` will now return a list of `RunComparison`s even if there is no baseline, whereas before it would have been an empty list? Is there something useful in a `RunComparison` with a contender but no baseline?

Reply:
Ah, rereading I think in that case we still use it to carry around the contender's sha, got it

Reply:
Good question because this is what triggered this whole refactor -- basically, in #20, I'll need to know all the runs for a given commit that had errors. Some of those runs might not have baseline runs but we still want to know their run IDs (so we can link to them) and the error tracebacks. I'll make a PR for #20 next.